File: microservices/backend-nuxt-krakenD/src/main/java/ea/ciges/krakend/model/Producto.java
package ea.ciges.krakend.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "PRODUCTO")
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Producto extends EntidadInfo<Long> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @NotBlank(message = "El nombre no puede estar vacío.")
    private String nombre;


    @NotBlank(message = "La descripción no puede estar vacía.")
    private String descripcion;

    private BigDecimal precio;

    private Boolean enStock;


}

File: nuxt3_app/model/Producto.ts
import { EntidadInfo } from "./EntidadInfo";

export default class Producto extends EntidadInfo{
    private id?: number;
    private nombre?: string;
    private descripcion?: string;
    private precio?: number;
    private enStock?: boolean;

    constructor(
        id?: number,
        nombre?: string,
        descripcion?: string,
        precio?: number,
        enStock?: boolean
    ) {
        super();
        this.id = id;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.enStock = enStock;
    }

    toJson(): string {
        return JSON.stringify({
            id: this.id,
            nombre: this.nombre,
            descripcion: this.descripcion,
            precio: this.precio,
            enStock: this.enStock,
        });
    }
}
